spring-rabbit/src/main/java/org/springframework/amqp/rabbit/annotation/Argument.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.amqp.rabbit.annotation;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Represents an argument used when declaring queues etc within a
+ * {@code QueueBinding}.
+ *
+ * @author Gary Russell
+ * @since 1.6
+ *
+ */
+@Target({})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface Argument {
+
+	/**
+	 * @return the argument name.
+	 */
+	String name() default "";
```

Comment:
I think `name` must be required.
Otherwise this annotation does not make sense.